### DISCUSSION TITLE
Does your lsp implementation support lsp over tcp?

### DISCUSSION THREAD

**Discussion by ghost** [SpartanJ/ecode#84] [Q&A] [answered]
This language server only operates over TCP:

https://github.com/objeck/objeck-lsp

Currently there are only two options to use with it as I know: Sublime Text or VSCode :face_exhaling:

**Comment by SpartanJ** [comment, marked as answer]
Hi! There's currently no support for TCP in the LSP Client, but I was thinking of adding it since it shouldn't be that hard and could expand the support spectrum of the LSP client. I'll take a look, I might add it soon.
Meanwhile, I'll add highlighting support for the language, since I see you already developed one for lite-xl. Also, I see Pascal too, cool. I'll add it too. And I'll add your name to the [acknowledgements](https://github.com/SpartanJ/ecode#acknowledgements).

**Reply by SpartanJ** [reply]
[Added](https://github.com/SpartanJ/eepp/commit/fae07f495d13f436398030c2d4a419ce50504dcf) languages support for the next release. [Added](https://github.com/SpartanJ/ecode/commit/8bb54a3125691fcc91d6f63fd564c83e7408185e) acknowledgement.

**Reply by SpartanJ** [reply]
Meanwhile you can use the JSON definitions.
[languages.zip](https://github.com/SpartanJ/ecode/files/10975112/languages.zip)

**Reply by SpartanJ** [reply]
@iqskpduswupkcjqg I added TCP support for the LSP client, it will be available in the next release. I tested it with well-established LSPs and works well, but, I tried the `objeck-lsp` and I couldn't get it to work correctly, it seems to crash with any document edit, I don't know if this is a problem with the Linux server or it's general and it's really unstable. But the LSP opens and answers the client's requests.
